include/swift/Frontend/ModuleInterfaceLoader.h

```diff
@@ -311,6 +311,9 @@ struct ModuleInterfaceLoaderOptions {
     case FrontendOptions::ActionType::TypecheckModuleFromInterface:
       requestedAction = FrontendOptions::ActionType::Typecheck;
       break;
+    case FrontendOptions::ActionType::ScanDependencies:
+      requestedAction = Opts.RequestedAction;
+      break;
```

Comment:
@nkcsgexi @artemcm Mind double checking that there aren't other callers of `CompilerInstance::setup` where we may need to change the requested action to avoid loading the stdlib? In particular I'm not too sure what arguments are being fed for the other dependency scanning call-sites, are they guaranteed to use the dependency scanning action?

Reply:
This looks fine to me.
I also fully expect all dependency scanning call-sites to also have `FrontendOptions::ActionType::ScanDependencies`.